crates/h264/README.md

```diff
@@ -7,7 +7,231 @@
+// p = 134
+let _ = writer.write_exp_golomb(134);
+
+// frame_mbs_only_flag
+let _ = writer.write_bit(true);
+
+// direct_8x8_inference_flag
+let _ = writer.write_bit(false);
+// frame_cropping_flag
+let _ = writer.write_bit(false);
+
+// vui_parameters_present_flag
+let _ = writer.write_bit(true);
+
+// enter vui to set the framerate
+// aspect_ratio_info_present_flag
+let _ = writer.write_bit(true);
+// we want square (1:1) for 16:9 for 4k w/o overscan
+// aspect_ratio_idc
+let _ = writer.write_bits(1, 8);
+
+// overscan_info_present_flag
+let _ = writer.write_bit(true);
+// we dont want overscan
+// overscan_appropriate_flag
+let _ = writer.write_bit(false);
+
+// video_signal_type_present_flag
+let _ = writer.write_bit(false);
+// chroma_loc_info_present_flag
+let _ = writer.write_bit(false);
+
+// timing_info_present_flag
+let _ = writer.write_bit(true);
+// we can set this to 100 for example
+// num_units_in_tick is a u32
+let _ = writer.write_bits(100, 32);
+// fps = time_scale / (2 * num_units_in_tick)
+// since we want 144 fps:
+// 144 = time_scale / (2 * 100)
+// 28800 = time_scale
+// time_scale is a u32
+let _ = writer.write_bits(28800, 32);
+let _ = writer.finish();
+
+// Now result contains a complete SPS.
+let result = Sps::parse(sps.into()).unwrap();
+```
+For more SPS examples, check out the tests in the source code for SPS.
+
+### Demuxing
+
+```rust
+use std::io::{self, Write};
+
+use bytes::Bytes;
+use scuffle_bytes_util::BitWriter;
+
+use scuffle_h264::{Sps, AVCDecoderConfigurationRecord};
+
+let mut data = Vec::new();
+let mut writer = BitWriter::new(&mut data);
+
+// configuration_version
+let _ = writer.write_bits(1, 8);
+// profile_indication
+let _ = writer.write_bits(100, 8);
+// profile_compatibility
+let _ = writer.write_bits(0, 8);
+// level_indication
+let _ = writer.write_bits(31, 8);
+// length_size_minus_one
+let _ = writer.write_bits(3, 8);
+
+// num_of_sequence_parameter_sets
+let _ = writer.write_bits(1, 8);
+// sps_length
+let _ = writer.write_bits(29, 16);
+// You can even pass an SPS as a byte string!
+// SPS
+let _ = writer.write_all(b"gd\0\x1f\xac\xd9A\xe0m\xf9\xe6\xa0  (\0\0\x03\0\x08\0\0\x03\x01\xe0x\xc1\x8c\xb0");
+
+// num_of_picture_parameter_sets
+let _ = writer.write_bits(1, 8);
+// pps_length
+let _ = writer.write_bits(6, 16);
+// You can also pass a PPS as a byte string!
+// pps
+let _ = writer.write_all(b"h\xeb\xe3\xcb\"\xc0\x00\x00");
+
+// chroma_format_idc
+let _ = writer.write_bits(1, 8);
+// bit_depth_luma_minus8
+let _ = writer.write_bits(0, 8);
+// bit_depth_chroma_minus8
+let _ = writer.write_bits(0, 8);
+// number_of_sequence_parameter_set_ext
+let _ = writer.write_bits(0, 8);
+let _ = writer.finish();
+
+// A demuxed config.
+let result = AVCDecoderConfigurationRecord::demux(&mut io::Cursor::new(data.into())).unwrap();
+
+// Do something with it!
+
+// You can also access the sps bytestream and parse it:
+let sps = &result.sps[0];
+let sps = Sps::parse(sps.clone()).unwrap();
+```
```

Comment:
I think, we only need an example on the demux where the data is raw bytes do not construct it via a bit writer.